syntax = "proto3";

package cafm.identity.user_group;

import "user_group.proto";

message UserGroupAppService_userGroupByNameRequest {
  string name = 1;
}

message UserGroupAppService_userGroupByNameResponse {
  UserGroup userGroup = 1;
}

message UserGroupAppService_userGroupByIdRequest {
  string id = 1;
}

message UserGroupAppService_userGroupByIdResponse {
  UserGroup userGroup = 1;
}

message UserGroupAppService_userGroupsRequest {
  int32 resultFrom = 1;
  int32 resultSize = 2;
}

message UserGroupAppService_userGroupsResponse {
  repeated UserGroup userGroups = 1;
}


service UserGroupAppService {
  rpc userGroupByName (UserGroupAppService_userGroupByNameRequest) returns (UserGroupAppService_userGroupByNameResponse) {}
  rpc userGroupById (UserGroupAppService_userGroupByIdRequest) returns (UserGroupAppService_userGroupByIdResponse) {}
  rpc userGroups (UserGroupAppService_userGroupsRequest) returns (UserGroupAppService_userGroupsResponse) {}
}